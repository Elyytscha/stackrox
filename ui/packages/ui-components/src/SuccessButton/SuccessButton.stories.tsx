import React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';

import SuccessButton from './SuccessButton';

export default {
    title: 'SuccessButton',
    component: SuccessButton,
} as Meta;

export const BasicUsage: Story = () => {
    function onClick(): void {}
    return (
        <SuccessButton type="button" onClick={onClick}>
            Save the Planet
        </SuccessButton>
    );
};
