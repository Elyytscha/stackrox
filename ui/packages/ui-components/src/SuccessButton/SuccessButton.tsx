import React, { ReactElement } from 'react';

import Button, { HOCButtonProps } from '../Button';

function SuccessButton({
    type = 'button',
    onClick,
    children,
    isDisabled = false,
}: HOCButtonProps): ReactElement {
    return (
        <Button colorType="success" type={type} onClick={onClick} isDisabled={isDisabled}>
            {children}
        </Button>
    );
}

export default SuccessButton;
