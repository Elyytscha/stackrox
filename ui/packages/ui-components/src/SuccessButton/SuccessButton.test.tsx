import React from 'react';
import { render } from '@testing-library/react';

import SuccessButton from './SuccessButton';

describe('SuccessButton', () => {
    it('renders', () => {
        function onClick(): void {}

        const { getByText } = render(
            <SuccessButton type="button" onClick={onClick}>
                Save cluster
            </SuccessButton>
        );

        expect(getByText('Save cluster')).toBeInTheDocument();
    });

    it('has the correct style', () => {
        function onClick(): void {}

        const { container } = render(
            <SuccessButton type="button" onClick={onClick}>
                Click me!
            </SuccessButton>
        );

        expect(container).toMatchSnapshot();
    });
});
