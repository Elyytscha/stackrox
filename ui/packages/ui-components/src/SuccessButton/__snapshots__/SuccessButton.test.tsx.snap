// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessButton has the correct style 1`] = `
<div>
  <button
    class="border-2 font-600 inline-flex items-center justify-center rounded-sm uppercase text-sm border-success-500 bg-success-200 hover:bg-success-300 hover:text-success-800 text-success-700 p-2"
    type="button"
  >
    Click me!
  </button>
</div>
`;
